docs/proposals/choosing-scheduler.md

```diff
@@ -0,0 +1,165 @@
+
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+
+<h2>PLEASE NOTE: This document applies to the HEAD of the source tree</h2>
+
+If you are using a released version of Kubernetes, you should
+refer to the docs that go with that version.
+
+<strong>
+The latest release of this document can be found
+[here](http://releases.k8s.io/release-1.1/docs/proposals/choosing-scheduler.md).
+
+Documentation for other releases can be found at
+[releases.k8s.io](http://releases.k8s.io).
+</strong>
+--
+
+<!-- END STRIP_FOR_RELEASE -->
+
+<!-- END MUNGE: UNVERSIONED_WARNING -->
+
+# Choosing scheduler in a multi-scheduler configuration
+
+## Introduction
+
+This document proposes a mechanism for assigning a scheduler to a pod
+in a system which is using multiple schedulers as described [here](multiple-schedulers.md).
+In particular, we decribe an admission controller which adds the scheduler name annotation,
+and which takes its configuration from a new API object called `PodPolicy` that determines,
+for each pod, which scheduler is responsible for scheduling that pod, based on various
+characteristics of the pod.
+
+## Implementation
+
+We will create a new admission controller called PodSchedulerPolicy. It will be configured
+using a new API object called `PodPolicy`. We have chosen a generic name and API
+for this object because we want to use the same object for defining other policies based on pod
+characteristics, for example see #17097.
+
+PodSchedulerPolicy will iterate through the `Rules` of the `PodPolicy` and apply the
+first one that matches the pod. Each `Rule` is a string specifying the scheduler name
+to apply to the pod if the pod satisfies the rule's PolicyPredicate, which currently is
+just a wrapper over `PodSelector`.
+
+## API
+
+### PodPolicy
+
+```go
+// A PodPolicy is a list of rules that define the policy. Each rule is tested
+// in order, and the first one that matches is applied.
+type PodPolicy struct {
+	Rules []PodPolicyRule `json:"rules,omitempty"`
+}
+
+type PodPolicyRule struct {
+	// The rule defined by Policy is applied if the pod satisfies all of the criteria in PolicyPredicate
+	// (currently this just means satisfying the PodSelector in PolicyPredicate).
+	// The reason we use PolicyPredicate instead of just putting *PodSelector here is to allow
+	// us to cleanly add additional predicates in the future; they would be added to
+	// PolicyPredicate, allowing PodPolicyRule to stay simple.
+	PolicyPredicate *PolicyPredicate `json:"policyPredicate,omitempty"`
+	Policy string `json:"policy"`
```

Comment:
> What values can this take?

My intent is that it can take any string. Interpreting the string is the purview of the component that is using a PodPolicy. For example, for the puts-a-scheduler-name-annotation-on-a-pod admission controller, this string is the scheduler name to apply. In another component, the string my be effectively an enum value, which triggers arbitrary behavior based on the value.

> Can you show somewhere what the YAML would look like for setting a named scheduler as default scheduler? 

Empty PodSelector matches all pods (in the namespace), so you could have the last PodPolicyRule in the list have a PolicyPredicate with an empty PodSelector; since rules are evaluated in order, this would have the effect of using the corresponding Policy for any pods that don't match any of the "real" rules.

> And suggest where this "intentionally generic" object would be extended to, for example, include network policy.

Sorry, my claims of "intentionally generic" are perhaps a bit overblown. My thought was simply that the "Policy" string could be used however the consumer wants; in the scheduler-chooser it's the name of the desired scheduler, while in another consumer it could be effectively an enum value. I didn't give any real thought to use cases outside of scheduler-chooser, I just kinda assumed a string would be good enough. I'm quite willing to believe that assumption is wrong. (BTW @bgrant0607 is the one who suggested to do something generic that could be reused by other components that need policies.)